control-plane/subcommand/install-cni/command_test.go

```diff
@@ -1,81 +1,129 @@
+	require.Equal(t, cmd.flagKubeconfig, config.DefaultKubeconfig)
+	require.Equal(t, cmd.flagLogLevel, config.DefaultLogLevel)
+	require.Equal(t, cmd.flagLogJSON, defaultLogJSON)
+	require.Equal(t, cmd.flagMultus, config.DefaultMultus)
+}
+
+func TestRun_DirectoryWatcher(t *testing.T) {
+	// Create a default configuration that matches golden file.
+
+	consulConfig := config.NewDefaultCNIConfig()
+	configFile := "10-kindnet.conflist"
+	baseConfigFile := "testdata/10-kindnet.conflist"
+	goldenFile := "testdata/10-kindnet.conflist.golden"
+	notLastConfigFile := "testdata/10-kindnet.conflist.notlast"
+
+	// Create a Command and context.
+	var err error
+	ctx, cancel := context.WithCancel(context.Background())
+	tempDir := t.TempDir()
+
+	// Setup the Command.
+	ui := cli.NewMockUi()
+	cmd := &Command{
+		UI: ui,
 	}
+	cmd.init()
+	cmd.logger, err = common.Logger("info", false)
+	require.NoError(t, err)
+
+	// Create the file watcher.
+	go func() {
+		err := cmd.directoryWatcher(ctx, consulConfig, tempDir, "")
+		require.NoError(t, err)
+	}()
+	time.Sleep(50 * time.Millisecond)
+
+	// Copy a base config file that does not contain the consul entry.
+	err = copyFile(baseConfigFile, tempDir)
+	require.NoError(t, err)
+	time.Sleep(50 * time.Millisecond)
+	// The golden file contains the consul config.
+	expected, err := ioutil.ReadFile(goldenFile)
+	require.NoError(t, err)
+	// Get the name of the config file in the tempDir and read it.
+	tempDestFile := filepath.Join(tempDir, configFile)
+	actual, err := ioutil.ReadFile(tempDestFile)
+	require.NoError(t, err)
+	// Filewatcher should have detected a change and appended to the config file. Make sure
+	// files match.
+	require.Equal(t, string(expected), string(actual))
+
+	// Event 2: config file changed where consul is not last in the plugin list.
+	err = replaceFile(notLastConfigFile, filepath.Join(tempDir, configFile))
+	require.NoError(t, err)
+	time.Sleep(50 * time.Millisecond)
+	// Re-read the config file so we can compare the updated config file.
+	actual, err = ioutil.ReadFile(tempDestFile)
+	require.NoError(t, err)
+	// Filewatcher should have detected change, fixed and appended to the config file. Make sure
+	// files match.
+	require.Equal(t, string(expected), string(actual))
 
-	// set context so that the command will timeout
-
-	// Create a default config
-	cfg := &config.CNIConfig{
-		Name:       pluginName,
-		Type:       pluginType,
-		CNIBinDir:  defaultCNIBinDir,
-		CNINetDir:  defaultCNINetDir,
-		Multus:     defaultMultus,
-		Kubeconfig: defaultKubeconfig,
-		LogLevel:   defaultLogLevel,
+	// Event 3: consul config was removed from the config file. Should detect and fix.
+	err = replaceFile(baseConfigFile, filepath.Join(tempDir, configFile))
+	require.NoError(t, err)
+	time.Sleep(50 * time.Millisecond)
+	// Filewatcher should have detected change, fixed and appended to the config file. Make sure
+	// files match.
+	require.Equal(t, string(expected), string(actual))
+	cancel()
```

Comment:
I think this would need to be in t.Cleanup in the beginning of the test somewhere
```suggestion
       t.Cleanup(cancel)
```

Reply:
Done.